### DISCUSSION TITLE
mAP while training

### DISCUSSION THREAD

**Discussion by MohamedDonia** [ultralytics/yolov5#5841] [General]
I want to calculate mAP over training data while training every batch

**Comment by glenn-jocher** [comment]
@MohamedDonia mAP is evaluated on your validation set during training after every epoch.

**Reply by MohamedDonia** [reply]
I want to calculate mAP on  training batch to monitor mAP while training  @glenn-jocher

**Reply by glenn-jocher** [reply]
@MohamedDonia I don't know what you mean by training batch, but it's obviously bad practice to evaluate metrics on your trianing data. You evaluate on a validation set to observe the generalization capability on the model.

**Reply by MohamedDonia** [reply]
@glenn-jocher  during looping every batch while training , i want to use predictions out of the model  to get mAP to visualize it during training  like losses. 
I know that after each epoch  model calculate   mAP over validation set
